src/test/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/prison/prisonersearch/PrisonerSearchApiClientTest.kt

```diff
@@ -106,4 +107,27 @@ class PrisonerSearchApiClientTest {
       postRequestedFor(urlEqualTo("/prisoner-search/prisoner-numbers"))
     )
   }
+
+  @Test
+  fun `successfully get prisoner details`() {
+    mockServer.stubGetPrisoner(200)
+    val result = client.getPrisoner(MatchPrisonerRequest("identifier"))
+
+    assertThat(result).isEqualTo(
+      listOf(
+        Prisoner(
+          firstName = "JIM",
+          lastName = "SMITH",
+          dateOfBirth = LocalDate.of(1991, 7, 31),
+          prisonerNumber = "A1278AA",
+          pncNumber = "1234/1234589A",
+          croNumber = "1234/111111",
+        )
+      )
+    )
+
+    mockServer.verify(
+      postRequestedFor(urlEqualTo("/prisoner-search/match-prisoners"))
```

Comment:
call to /prisoners/{prisonNumber} and /prisoner-search/match-prisoners both hit the same endpoint  in prisoner-offender-search service. However each call needs to responds with data  having a different structure hence the reason for creating the Prisoner object